datumaro/plugins/data_formats/segment_anything/exporter.py

```diff
@@ -0,0 +1,187 @@
+from datumaro.components.exporter import Exporter
+from datumaro.components.media import Image
+from datumaro.util import NOTSET
+from datumaro.util import annotation_util as anno_tools
+from datumaro.util import dump_json_file, mask_tools
+
+
+def replace(json_data, key, value_new):
+    value_origin = json_data[key]
+    if value_origin is NOTSET:
+        json_data[key] = value_new
+        return
+    if value_origin != value_new:
+        raise DatumaroError(f"The value for '{key}' is not same for item {value_new}")
+
+
+class SegmentAnythingExporter(Exporter):
+    DEFAULT_IMAGE_EXT = ".jpg"
+
+    _polygon_types = {AnnotationType.polygon, AnnotationType.ellipse}
+    _allowed_types = {
+        AnnotationType.bbox,
+        AnnotationType.polygon,
+        AnnotationType.mask,
+        AnnotationType.ellipse,
+    }
+
+    def __init__(
+        self,
+        extractor,
+        save_dir,
+        **kwargs,
+    ):
+        super().__init__(extractor, save_dir, **kwargs)
+
+    @staticmethod
+    def find_instance_anns(annotations):
+        return [a for a in annotations if a.type in SegmentAnythingExporter._allowed_types]
+
+    @classmethod
+    def find_instances(cls, annotations):
+        return anno_tools.find_instances(cls.find_instance_anns(annotations))
+
+    def get_annotation_info(self, group, img_width, img_height):
+        boxes = [a for a in group if a.type == AnnotationType.bbox]
+        polygons: List[Union[Polygon, Ellipse]] = [
+            a for a in group if a.type in self._polygon_types
+        ]
+        masks = [a for a in group if a.type == AnnotationType.mask]
+
+        anns = boxes + polygons + masks
+        leader = anno_tools.find_group_leader(anns)
+        if len(boxes) > 0:
+            bbox = anno_tools.max_bbox(boxes)
+        else:
+            bbox = anno_tools.max_bbox(anns)
+        polygons = [p.as_polygon() for p in polygons]
+
+        mask = None
+        if polygons:
+            mask = mask_tools.rles_to_mask(polygons, img_width, img_height)
+        if masks:
+            masks = (m.image for m in masks)
+            if mask is not None:
+                masks = chain(masks, [mask])
+            mask = mask_tools.merge_masks(masks)
+        if mask is None:
+            return None
+        mask = mask_tools.mask_to_rle(mask)
+
+        segmentation = {
+            "counts": list(int(c) for c in mask["counts"]),
+            "size": list(int(c) for c in mask["size"]),
+        }
+        rles = mask_utils.frPyObjects(segmentation, img_height, img_width)
+        if isinstance(rles["counts"], bytes):
+            rles["counts"] = rles["counts"].decode()
+        area = mask_utils.area(rles)
+
+        annotation_data = {
+            "id": leader.group,
+            "segmentation": rles,
+            "bbox": bbox,
+            "area": area,
+            "predicted_iou": 0.0,
+            "stability_score": 0.0,
+            "crop_box": [],
+            "point_coords": [],
```

Comment:
```suggestion
     point_coords = set(
          point_coord
          for ann in anns
          for point_coord in ann.attributes.get("point_coords", [[]])
     )
     ...
            "point_coords": point_coords,
```

Reply:
Fixed in https://github.com/openvinotoolkit/datumaro/pull/1005/commits/161f31fccd9a5f7edbb853bb9f95d331d1709bfb